src/build.sh

```diff
@@ -1,21 +1,36 @@
 #!/usr/bin/env bash
 set -eu
 
-die() {
-    echo "*** ERROR; build failed"
-    exit 1
-}
+#===============================================================================
+# Assuming this script is run with: `cd $GERBIL_BASE/src && ./build.sh`
+# All other build scripts depend on this point of entry
+#===============================================================================
+
+## global (for all build scripts) setup
+readonly GERBIL_SOURCE=$(pwd -P)
+readonly GERBIL_BASE="$(dirname "${GERBIL_SOURCE}")"
+readonly BUILD_SCRIPT_DIR="${GERBIL_SOURCE}/build_scripts"
+readonly GERBIL_SETUP="" # this flag signals that basic setup has occured
+
+export GERBIL_SOURCE
+export GERBIL_BASE
+export BUILD_SCRIPT_DIR
+export GERBIL_SETUP
 
+## common
+source "${BUILD_SCRIPT_DIR}/common.sh"
+
+## main
 build_gerbil() {
-    ./build0.sh || die
-    ./build1.sh final || die
-    ./build2_fini.sh
-    ./build_stdlib.sh || die
-    ./build_lang.sh || die
-    ./build_tools.sh || die
+    "${BUILD_SCRIPT_DIR}/build0.sh"       || die
+    "${BUILD_SCRIPT_DIR}/build1.sh" final || die
+    # TODO: if I understand correctly, 'final' should call the cleanup script
```

Comment:
it's a historical arifact from when there was a stage2 build; 
we can remove it and make the logic in build1 do final build by default.